File: leetcode/php/1646-get-maximum-in-generated-array.php
<?php

/**
 * 1646. Get Maximum in Generated Array
 * https://leetcode.com/problems/get-maximum-in-generated-array/
 *
 * https://leetcode.com/problems/get-maximum-in-generated-array/discuss/931123/PHP-Easy-and-short-O(n)-solution
 */

class Solution
{
    /**
     * @param Integer $n
     * @return Integer
     */
    function getMaximumGenerated($n)
    {
        for ($array = [0], $i = 1; $i <= $n; $i++)
            $array[] = $i % 2 ? ($array[($i - 1) / 2] + ($array[($i + 1) / 2] ?? 1)) : $array[$i / 2];
        return max($array);
    }
}

File: leetcode/swift/1646-get-maximum-in-generated-array.swift
/**
 * 1646. Get Maximum in Generated Array
 * https://leetcode.com/problems/get-maximum-in-generated-array/
 */

class Solution {
    func getMaximumGenerated(_ n: Int) -> Int {
        var arr: [Int] = [0, 1], i = 1
        while i < n {
            i += 1
            arr.append(i % 2 == 0 ? arr[i/2] : arr[(i-1)/2] + arr[(i+1)/2])
        }
        return n < 1 ? 0 : arr.max()!
    }
}